scripts/tslint/enforceConsecutiveFunctionOverloadRule.ts

```diff
@@ -0,0 +1,46 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+
+    public static FAILURE_STRING_FACTORY = (name: string) => `All '${name}' signatures should be adjacent`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoNonConsecutiveFunctionOverload(sourceFile, this.getOptions()));
+    }
+}
+
+class NoNonConsecutiveFunctionOverload extends Lint.BlockScopeAwareRuleWalker<{}, ScopeInfo> {
+
+    createScope(): any {
+        return undefined;
+    }
+
+    createBlockScope(): ScopeInfo {
+        return new ScopeInfo();
+    }
+
+    visitInterfaceDeclaration(node: ts.InterfaceDeclaration): void {

```

Comment:
Actually, we need to check those in declaration files. See palantir/tslint#1707.